sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -419,55 +442,51 @@ public static ExtractKeyPhrasesResultCollection toExtractKeyPhrasesResultCollect
+                    IterableStream.of(healthcareEntities));
 
-                Map<HealthcareEntity, Map<HealthcareEntity, HealthcareEntityRelationType>> entityRelationMap =
-                    new HashMap<>();
-                if (!CoreUtils.isNullOrEmpty(documentEntities.getRelations())) {
-                    documentEntities.getRelations().forEach(healthcareRelation -> {
-                        // TODO: bidirectional relation is gone, do it in another PR
-                        // issue: https://github.com/Azure/azure-sdk-for-java/issues/19180
-//                        final HealthcareEntity targetEntity =
-//                            healthcareEntities.get(getHealthcareEntityIndex(healthcareRelation.getTarget()));
-//                        final HealthcareEntity sourceEntity =
-//                            healthcareEntities.get(getHealthcareEntityIndex(healthcareRelation.getSource()));
-//                        final HealthcareEntityRelationType relationType =
-//                            HealthcareEntityRelationType.fromString(healthcareRelation.getRelationType());
-
-//                        final Map<HealthcareEntity, HealthcareEntityRelationType> targetRelatedEntity =
-//                            entityRelationMap.getOrDefault(targetEntity, new HashMap<>());
-//                        targetRelatedEntity.putIfAbsent(sourceEntity, relationType);
-//                        entityRelationMap.putIfAbsent(targetEntity, targetRelatedEntity);
-//
-//                        if (healthcareRelation.isBidirectional()) {
-//                            final Map<HealthcareEntity, HealthcareEntityRelationType> sourceRelatedEntity =
-//                                entityRelationMap.getOrDefault(sourceEntity, new HashMap<>());
-//                            sourceRelatedEntity.putIfAbsent(targetEntity, relationType);
-//                            entityRelationMap.putIfAbsent(sourceEntity, sourceRelatedEntity);
-//                        }
-                    });
+                final List<HealthcareRelation> relations = documentEntities.getRelations();
+                if (!CoreUtils.isNullOrEmpty(relations)) {
+                    final List<HealthcareEntityRelation> entityRelations = relations.stream().map(
+                        healthcareRelation -> {
+                            final HealthcareEntityRelation entityRelation = new HealthcareEntityRelation();
+                            final RelationType relationType = healthcareRelation.getRelationType();
+                            if (relationType != null) {
+                                HealthcareEntityRelationPropertiesHelper.setRelationType(entityRelation,
+                                    HealthcareEntityRelationType.fromString(relationType.toString()));
+                            }
+
+                            final List<HealthcareRelationEntity> relationEntities = healthcareRelation.getEntities();
```

Comment:
Can healthcareRelation.getEntities() result in being null?

Reply:
Yes. It could be null. I will update the check for null and empty list.

Reply:
If I understand the code, you are accessing `HealthcareRelation.Entities` which according to the [swagger](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json#L1819), is required, so no chance it can be null.

Reply:
I think might be we should always return an empty list to user by default other than return null to them. Except null have a meaning.

Reply:
Just saw the message that above my latest comment, thanks @maririos for the correction.